src/iface/ethernet.rs

```diff
@@ -679,6 +703,36 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
         }
     }
 
+    #[cfg(feature = "proto-ipv6")]
+    fn process_icmpv6<'frame>(&self, _sockets: &mut SocketSet, ip_repr: IpRepr,
+                              ip_payload: &'frame [u8]) -> Result<Packet<'frame>>
+    {
+        let icmp_packet = Icmpv6Packet::new_checked(ip_payload)?;
+        let checksum_caps = self.device_capabilities.checksum.clone();
+        let icmp_repr = Icmpv6Repr::parse(&icmp_packet, &checksum_caps)?;
+
+        match icmp_repr {
+            // Respond to echo requests.
+            Icmpv6Repr::EchoRequest { ident, seq_no, data } => {
+                let icmp_reply_repr = Icmpv6Repr::EchoReply {
+                    ident:  ident,
+                    seq_no: seq_no,
+                    data:   data
+                };
+                match ip_repr {
+                    IpRepr::Ipv6(ipv6_repr) => Ok(self.icmpv6_reply(ipv6_repr, icmp_reply_repr)),
+                    _ => Err(Error::Unrecognized),
+                }
+            }
+
+            // Ignore any echo replies.
+            Icmpv6Repr::EchoReply { .. } => Ok(Packet::None),
+
+            // FIXME: do something correct here?
+            _ => Err(Error::Unrecognized),
```

Comment:
According to [IPv6 Essentials](https://www.safaribooksonline.com/library/view/ipv6-essentials/0596001258/):

> If a node receives an ICMPv6 informational message of unknown type, it must be silently discarded.

Reply:
Good to know. There are some known ICMPv6 error messages that could hit this. I think this FIXME is largely to pass ICMPv6 error messages to higher layers. (An issue should probably be created for this)